bpf/process/types/basic.h

```diff
@@ -560,10 +560,11 @@ filter_char_buf(struct selector_arg_filter *filter, char *args)
 		asm volatile("%[j] &= 0xff;\n" ::[j] "+r"(j)
 			     :);
 		length = *(__u32 *)&value[j];
-		asm volatile("%[length] &= 0x3f;\n" ::[length] "+r"(length)
+		asm volatile("%[length] &= 0x3f;\n" ::[length] "+r"(
+				     length)
 			     :);
-		v = (int)value[j];
-		a = (int)args[0];
+		v = ((int *)value)[j];
```

Comment:
While the original construction was broken (only taking the first byte of the length at &value[j], this replacement appears to incorrectly treat j as a (int *) index rather than a (char *) index. It probably needs to be:
v = *(int *)&value[j];
instead. Similar should be done for the following line, although it obviously matters less.

Reply:
👋 It looks like you've removed this change. I think you were right it could do with fixing (for future proofing it nothing else). Did you disagree with "v = *(int *)&value[j];" ?

Reply:
i think you are right, I agree "v = *(int *)&value[j];". I removed the change to determine whether it was impossible to pass the test case before the change